test/Sema/diag_erroneous_iuo.swift

```diff
@@ -116,9 +116,9 @@ _ = Array<Int!>() // expected-error {{'!' is not allowed here; perhaps '?' was i
 let _: Array<Int!> = [1] // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{17-18=?}}
 _ = [Int!]() // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{9-10=?}}
 let _: [Int!] = [1] // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{12-13=?}}
-_ = Optional<Int!>(nil) // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{17-18=?}}
+_ = Optional<Int!>(nil) // expected-error 2 {{'!' is not allowed here; perhaps '?' was intended?}}{{17-18=?}}
 let _: Optional<Int!> = nil // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{20-21=?}}
-_ = Int!?(0) // expected-error {{'!' is not allowed here; perhaps '?' was intended?}}{{8-9=?}}
+_ = Int!?(0) // expected-error 3 {{'!' is not allowed here; perhaps '?' was intended?}}{{8-9=?}}
```

Comment:
This looks like we're processing the type multiple times…

Reply:
Yeah, we have that in a couple of places already unfortunately, because we don't track if we produced the same diagnostic multiple times...

Reply:
Looking at the TypeLoc, we should see it has already been resolved to a Type, and not resolve the TypeRepr again. Where is the diagnostic being emitted?

Reply:
Ah I see, it's probably my calls to resolveType which caused this, I'll try to fix it.

Reply:
Done, just needed to cache the type in a couple of places :)